pkg/api/handlers/mcpgateway/oauth/mcpoauthhandler.go

```diff
@@ -38,18 +40,61 @@ func NewMCPOAuthHandlerFactory(baseURL string, sessionManager *mcp.SessionManage
 		tokenStore:        globalTokenStore,
 	}
 }
-func (f *MCPOAuthHandlerFactory) CheckForMCPAuth(ctx context.Context, mcpServer v1.MCPServer, mcpServerConfig mcp.ServerConfig, userID, mcpID, oauthAppAuthRequestID string) (string, error) {
-	if mcpServerConfig.Runtime != types.RuntimeRemote {
-		// OAuth is only support for remote MCP servers.
+func (f *MCPOAuthHandlerFactory) CheckForMCPAuth(req api.Context, mcpServer v1.MCPServer, mcpServerConfig mcp.ServerConfig, userID, mcpID, oauthAppAuthRequestID string) (string, error) {
+	switch mcpServerConfig.Runtime {
+	case types.RuntimeRemote:
+	case types.RuntimeComposite:
```

Comment:
I think this is a time where the switch statement is more confusing. I thought this was a bug at first.